src/QsCompiler/BondSchemas/QirExecutionWrapper/Protocols.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using Bond;
+using Bond.IO.Unsafe;
+using Bond.Protocols;
+
+namespace Microsoft.Quantum.QsCompiler.BondSchemas.QirExecutionWrapper
+{
+    /// <summary>
+    /// This class provides methods for serialization/deserialization of objects in the Microsoft.Quantum.QsCompiler.BondSchemas.SandboxInput namespace.
+    /// </summary>
+    internal static class Protocols
+    {
+        /// <summary>
+        /// Deserializes a sandbox input object from its fast binary representation.
+        /// </summary>
+        public static QirExecutionWrapper DeserializeFromFastBinary(Stream stream)
+        {
+            stream.Flush();
+            stream.Position = 0;
```

Comment:
When performing deserialization from Streams, I think it is a good idea to not perform any flush or change in position operation. The main reason is that it is the responsibility of the caller to do these operations. Another reason (which in our case is not that relevant) is that the caller might have multiple things in the same stream and it has set the position of the stream where it should start reading in order to deserialize a specific object.